workflow/schemas/samples.schema.yaml

```diff
@@ -8,6 +8,9 @@ properties:
   condition:
     type: string
     description: sample condition that will be compared during differential expression analysis (e.g. a treatment, a tissue time, a disease)
-
+  batch_effect:
+    type: string
+    description: The batch that the sample belongs to
+    
```

Comment:
As the column name used in the `config["diffexp"]["models"]["modelX"]` specifications of `full` and `reduced` models is arbitrary, so I'm not sure if we need to include this in the schema (even if we don't include among the `required:` fields listed below). Does specifying this here give you a clear benefit (like a more useful error message in some case) or was this just meant for consistency?

Reply:
It turned out not to be so arbitrary, I think that it should be enforced.

Reply:
Ah, but it should be. If you simply name such a column differently in the `samples.tsv` file, for example `day_of_processing`, you can simply adjust the formulas in `config.yaml` to:
```
full: ~condition + day_of_processing
reduced: ~day_of_processing
```

Or if you don't have any batch effects, you can also do something like:
```
full: ~condition
reduced: ~1
```

Is this explanation clearer?